pkg/ottl/ottlfuncs/func_community_id.go

```diff
@@ -0,0 +1,194 @@
+
+import (
+	"context"
+	//gosec:disable G505 -- SHA1 is intentionally used for generating unique identifiers
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type CommunityIDArguments[K any] struct {
+	SourceIP        ottl.StringGetter[K]
+	SourcePort      ottl.IntGetter[K]
+	DestinationIP   ottl.StringGetter[K]
+	DestinationPort ottl.IntGetter[K]
+	Protocol        ottl.Optional[ottl.StringGetter[K]]
+	Seed            ottl.Optional[ottl.IntGetter[K]]
+}
+
+func NewCommunityIDFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("CommunityID", &CommunityIDArguments[K]{}, createCommunityIDFunction[K])
+}
+
+func createCommunityIDFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*CommunityIDArguments[K])
+
+	if !ok {
+		return nil, errors.New("CommunityIDFactory args must be of type *CommunityIDArguments[K]")
+	}
+
+	return communityID(
+		args.SourceIP,
+		args.SourcePort,
+		args.DestinationIP,
+		args.DestinationPort,
+		args.Protocol,
+		args.Seed,
+	), nil
+}
+
+func communityID[K any](
+	sourceIP ottl.StringGetter[K],
+	sourcePort ottl.IntGetter[K],
+	destinationIP ottl.StringGetter[K],
+	destinationPort ottl.IntGetter[K],
+	protocol ottl.Optional[ottl.StringGetter[K]],
+	seed ottl.Optional[ottl.IntGetter[K]],
+) ottl.ExprFunc[K] {
+	return func(ctx context.Context, tCtx K) (any, error) {
+		srcIPValue, err := sourceIP.Get(ctx, tCtx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get source IP: %w", err)
+		}
+
+		dstIPValue, err := destinationIP.Get(ctx, tCtx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get destination IP: %w", err)
+		}
+
+		srcPort, err := sourcePort.Get(ctx, tCtx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get source port: %w", err)
+		}
+
+		dstPort, err := destinationPort.Get(ctx, tCtx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get destination port: %w", err)
+		}
+
+		protocolValue := uint8(6) // defaults to TCP
+		if !protocol.IsEmpty() {
+			protocolStr, err := protocol.Get().Get(ctx, tCtx)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get protocol: %w", err)
+			}
+
+			resolvedProtocolValue, err := resolveProtocolValue(protocolStr)
+			if err != nil {
+				return nil, err
+			}
+			protocolValue = resolvedProtocolValue
+		}
+
+		srcIPAddr := net.ParseIP(srcIPValue)
+		if srcIPAddr == nil {
+			return nil, fmt.Errorf("invalid source IP: %s", srcIPValue)
+		}
+
+		dstIPAddr := net.ParseIP(dstIPValue)
+		if dstIPAddr == nil {
+			return nil, fmt.Errorf("invalid destination IP: %s", dstIPValue)
+		}
+
+		// Get seed value (default: 0) if applied
+		seedValue := uint16(0)
+		if !seed.IsEmpty() {
+			seedInt, err := seed.Get().Get(ctx, tCtx)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get seed: %w", err)
+			}
+			seedValue = uint16(seedInt)
```

Comment:
I think this number conversion might overflow. The same for line 115.

Reply:
I have added a validation rule to make sure we get desired byte-size/value. Do you have a better idea?